src/components/Board/Board.js

```diff
@@ -50,63 +46,71 @@ const Board = () => {
-      // so they only take in the necessary information as a payload :)
+    // foundSnake(currentPlayerPos) is being called multiple times,It would be worth saving it into a variable and re-use it :)
+    //Done:)
+    const foundSnakeItem = foundSnake(currentPlayerPos);
+    const foundLadderItem = foundLadder(currentPlayerPos);
+    if (foundSnakeItem !== undefined) {
+      // I would also add that I think you can simplify most of your dispatch methods,so they only take in the necessary information as a payload :)
       // For instance: "snake" should only take in 1 argument :)
-      dispatch({
-        type: "snake",
-        payload: { isSnake: true, snake: foundSnake(currentPlayerPos) },
-      });
-      currentPlayerPos = foundSnake(currentPlayerPos).dest;
+      //Done:)
+      dispatch({ type: "snake", payload: foundSnakeItem });
+      currentPlayerPos = foundSnakeItem.dest;
       // Similar comments to this part of the function :)
-    } else if (foundLadder(currentPlayerPos) !== undefined) {
-      dispatch({
-        type: "ladder",
-        payload: { isLadder: true, ladder: foundLadder(currentPlayerPos) },
-      });
-      currentPlayerPos = foundLadder(currentPlayerPos).dest;
+      //Done:)
+    } else if (foundLadderItem !== undefined) {
+      dispatch({ type: "ladder", payload: foundLadderItem });
+      currentPlayerPos = foundLadderItem.dest;
     }
 
     return currentPlayerPos;
   };
 
-  // This function simply reset the state of snake and ladder to false after each roll.
-  // If each roll should reset the snake and ladder,
-  // then you should probably include that logic into that action :)
-  const resetSnakeAndLadderStates = () => {
-    if (snake) dispatch({ type: "snake", payload: { isSnake: false } });
-    if (ladder) dispatch({ type: "ladder", payload: { isLadder: false } });
-  };
-
   // What this function does:
   //1. Check if the player is allowed to start the game(if it has thrown a 6?).
   //2. If so, if the player position + random number <=100 (the max number on the board), it sets an Interval to increase the player's position one by one(to demonstrate the player moving on the board).
   //3. It sets a Timeout to terminate this process by clearing the interval, and checking if the new player's position is a snake or ladder. (by calling the 'applySnakeOrLadder' function)
   //4. Then updates the player's position and changes the turn.
   //5. If the player is not allowed so far to start the game, first, it checks if they just got a 6, and then update the player's permission; and then it changes the turn anyway.
 
+  const rollClick = () => {
+    // Hmmm this action could exist without a payload ;)
+    //******************************************* QUESTION ********************************************
+    //I'm using the same random number as an argument for the setPlayerNewPos() method. So I cannot generate the random number in the reducer file.
+
+    let randomNum = createRandomNumber();
```

Comment:
hmm you might be right 🤔 
I will play with the code a little bit :)

Reply:
Here was my first struggle with the code, if you remember the very first question that I asked you was "How to update a state and **THEN** do something else". You remember our conversion about states not returning promises after getting updated?
I wanted to update the die state in the reducer file(as you might be thinking about), and **THEN** update the position of players based on that.